lib/puppet/settings/server_list_setting.rb

```diff
@@ -4,6 +4,10 @@ def type
     :server_list
   end
 
+  def print(value)
+    value
```

Comment:
After having some time to let this percolate in the back of my mind for a while, I'm wondering whether or not it would be a good idea to have the print method try to convert the nested arrays back to the "stringy" version (Eg: `[["myserver", 123], ["yourserver", 456]]` `->` `"myserver:123,yourserver:456"`). One of the big things making me wonder about how useful this would actually be though, is that it's kinda hard to track down whether the print method seeing the array of arrays version is purely a testing artifact, or whether it's likely to happen in the wild.

As a complete asside: Having an instance method where you need to pass in value of one of the properties of the instance that you're calling the method on is pretty gross.

Reply:
Maybe I'm misunderstanding, but the arrays of arrays occurs when puppet.conf contains:

```
[agent]
server_list = neptune, mars
```

and you try to print `server_list`:

```
$ bx puppet config print server_list --section agent
[["neptune"], ["mars"]]
```

About the aside, yes unfortunately each setting object doesn't own the value of the setting. It just knows how to interpret the value, which is stored in the global `Puppet::Settings` object. I agree it's gross.